### PR TITLE
some fixes

### PR DESCRIPTION
**Pull request by darthorimar**

### CODE REVIEW

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/InstanceofConversion.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import org.jetbrains.kotlin.j2k.tree.impl.*
+import org.jetbrains.kotlin.psi.KtClass
+
+
+class InstanceofConversion : RecursiveApplicableConversionBase() {
+    override fun applyToElement(element: JKTreeElement): JKTreeElement {
+        if (element is JKJavaInstanceOfExpression) {
+            val checkingType = element.type.type
+            val type =
+                when (checkingType) {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/InstanceofConversion.kt]: Join this when with if on 24

File: j2k/newSrc/org/jetbrains/kotlin/j2k/NewCodeBuilder.kt [hunk 2]
```diff
@@ -361,6 +361,17 @@ class NewCodeBuilder {
             }
         }
 
+        override fun visitJavaNewExpression(javaNewExpression: JKJavaNewExpression) {
+            printer.printWithNoIndent(FqName(javaNewExpression.constructorSymbol.fqName).shortName())//TODO import name somehow?
+            printer.printWithNoIndent("(")
+            //TODO use new sequence generator in future
+            for ((index, argument) in javaNewExpression.arguments.expressions.withIndex()) {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/NewCodeBuilder.kt]: renderList

File: j2k/newSrc/org/jetbrains/kotlin/j2k/NewCodeBuilder.kt [hunk 3]
```diff
@@ -94,7 +94,10 @@ class NewCodeBuilder {
             )
         }
 
-        override fun visitMutabilityModifier(mutabilityModifier: JKMutabilityModifier) {}
+        override fun visitMutabilityModifier(mutabilityModifier: JKMutabilityModifier) {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/NewCodeBuilder.kt:152]: Fix also in `visitKtProperty`

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/FieldInitializersInPrimaryFromParamsConversion.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+package org.jetbrains.kotlin.j2k.conversions
+
+import org.jetbrains.kotlin.j2k.tree.*
+import org.jetbrains.kotlin.j2k.tree.impl.JKJavaFieldImpl
+import org.jetbrains.kotlin.j2k.tree.impl.JKNameIdentifierImpl
+
+
+class FieldInitializersInPrimaryFromParamsConversion : TransformerBasedConversion() {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/FieldInitializersInPrimaryFromParamsConversion.kt]: Please, use stateful MatchBasedConversion instead

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/ModifiersConversion.kt [hunk 5]
```diff
@@ -24,18 +26,42 @@ class ModifiersConversion(private val context: ConversionContext) : RecursiveApp
...
             }
+            if (element is JKParameter) {
+                convertParameterModifiers(element)
+            }
         }
         return recurse(element)
     }
 
+    private fun convertParameterModifiers(jkParameter: JKParameter) {
+        jkParameter.filterModifiers {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/ModifiersConversion.kt]: TODO: open is allowed on property-parameters, maybe handle in printer?

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/ModifiersConversion.kt [hunk 6]
```diff
@@ -24,18 +26,42 @@ class ModifiersConversion(private val context: ConversionContext) : RecursiveApp
...
+            when (it) {
+                is JKModalityModifier -> it.modality != JKModalityModifier.Modality.OPEN
+                else -> true
+            }
+        }
+        if (jkParameter.parent is JKKtPrimaryConstructor) {
+            val hasNonDefaultMutabilityModifier =
+                jkParameter.modifierList.modifiers.find {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/ModifiersConversion.kt]: `any`

File: j2k/newSrc/org/jetbrains/kotlin/j2k/ConversionsRunner.kt [hunk 7]
```diff
@@ -32,7 +32,7 @@ object ConversionsRunner {
             +BlockToRunConversion(context)
             +ConstructorConversion(context)
             +PrimaryConstructorDetectConversion(context)
-            +InsertDefaultPrimaryConstructorConversion(context)
+//            +InsertDefaultPrimaryConstructorConversion(context)
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/ConversionsRunner.kt]: ----

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/InsertDefaultPrimaryConstructorConversion.kt [hunk 8]
```diff
@@ -10,6 +10,8 @@ import org.jetbrains.kotlin.j2k.tree.*
 import org.jetbrains.kotlin.j2k.tree.JKModalityModifier.Modality.*
 import org.jetbrains.kotlin.j2k.tree.impl.*
 
+
+//TODO not needed???
 class InsertDefaultPrimaryConstructorConversion(private val context: ConversionContext) : RecursiveApplicableConversionBase() {
     override fun applyToElement(element: JKTreeElement): JKTreeElement {
         if (element !is JKClass) return recurse(element)
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/InsertDefaultPrimaryConstructorConversion.kt:16]: Replace none with any on 18 to fix invalid application

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/FieldInitializersInPrimaryFromParamsConversion.kt [hunk 9]
```diff
@@ -29,8 +28,9 @@ class FieldInitializersInPrimaryFromParamsConversion : TransformerBasedConversio
             if (ktPrimaryConstructor.parameters.contains(parameter)) {
                 val fieldDeclaration = containingClass.declarationList.find {
                     (it as? JKField)?.name?.value == fieldTarget.name.value
-                } ?: continue
-                parameter.modifierList = (fieldDeclaration as JKField).modifierList.also { it.detach(it.parent!!) }
+                } as? JKField ?: continue
+                if (!fieldDeclaration.type.type.equalsByName(parameter.type.type)) continue//TODO better way to compare types??
+                parameter.modifierList = fieldDeclaration.modifierList.also { it.detach(it.parent!!) }
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/FieldInitializersInPrimaryFromParamsConversion.kt]: implement and use detached here

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/k.kt [hunk 10]
```diff
@@ -222,4 +222,12 @@ fun JKClass.getOrCreateInitDeclaration(): JKKtInitDeclaration {
     val newDeclaration = JKKtInitDeclarationImpl(JKBlockImpl())
     declarationList += newDeclaration
     return newDeclaration
+}
+
+class JKKtForInStatementImpl(variableIdentifier: JKNameIdentifier, iterationExpression: JKExpression, body: JKStatement) : JKKtForInStatement,
+    JKBranchElementBase() {
+    override var variableIdentifier: JKNameIdentifier by child(variableIdentifier)
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/k.kt]: Can be also a destructuring declaration, so maybe DeclarationStatement here?

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/LabeledStatementConversion.kt [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+        if (isConvertedForLoopToWhileOne(statement)) {
+            val initStatement = statement.block.statements[0].detached()
+            val whileStatement = statement.block.statements[1].detached()
+            val whileStatementLabeled = JKLabeledStatementImpl(whileStatement, element.labels.also { it.forEach { it.detached() } })
+            return recurse(JKBlockStatementImpl(JKBlockImpl(listOf(initStatement, whileStatementLabeled))))
+        }
+        return recurse(element)
+    }
+
+    private fun isConvertedForLoopToWhileOne(blockStatement: JKBlockStatement): Boolean =
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/LabeledStatementConversion.kt]: Special AST node for such converted elements

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt [hunk 12]
```diff
@@ -104,19 +109,57 @@ class JKBlockImpl(statements: List<JKStatement> = emptyList()) : JKBlock, JKBran
...
     override var operator: JKOperator
 ) : JKBinaryExpression,
     JKBranchElementBase() {
     override fun <R, D> accept(visitor: JKVisitor<R, D>, data: D): R = visitor.visitBinaryExpression(this, data)
-
     override var right by child(right)
     override var left by child(left)
+
+    companion object {
+        private fun methodSymbolForToken(
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt]: Extract somewhere

File: j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/ForConversion.kt [hunk 13]
```diff
@@ -97,7 +97,7 @@ class ForConversion(private val context: ConversionContext) : RecursiveApplicabl
             val start = loopVar.initializer
             val operationType =
                 (loopStatement.updater as? JKExpressionStatement)?.expression?.isVariableIncrementOrDecrement(loopVar)
-            val reversed = when ((operationType  as? JKJavaOperatorImpl)?.token) {
+            val reversed = when ((operationType  as? JKJavaOperatorImpl)?.token?.token) {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/conversions/ForConversion.kt]: psiToken maybe?

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/symbols.kt [hunk 14]
```diff
@@ -36,13 +36,14 @@ interface JKClassSymbol : JKNamedSymbol {
...
     val fqName: String
-    val recivierType: JKType?
+    val receiverType: JKType?
     val parameterTypes: List<JKType>
     val returnType: JKType
 }
 
 interface JKFieldSymbol : JKNamedSymbol {
     val fqName: String
+    val filedType: JKType
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/symbols.kt]: fieldType

**Review by semoro — CHANGES_REQUESTED** [review]

**Comment by darthorimar** [comment]
Closed due to https://github.com/semoro/kotlin/pull/22

**Pull request closed without merging**